lib/view_model/cake_pay/cake_pay_cards_list_view_model.dart

```diff
@@ -220,15 +207,16 @@ abstract class CakePayCardsListViewModelBase with Store {
     try {
       searchString = text ?? '';
       var newVendors = await cakePayService.getVendors(
-          country: Country.getCakePayName(selectedCountry),
+          countryCode: selectedCountry.countryCode,
           page: currentPage ?? page,
           search: searchString,
           giftCards: displayGiftCards,
-          prepaidCards: displayPrepaidCards,
-          custom: displayCustomValueCards,
-          onDemand: displayDenominationsCards);
+          prepaidCards: displayPrepaidCards);
 
-      cakePayVendors = CakePayVendorList = newVendors;
+      cakePayVendors = newVendors.where((vendor) {
+        if (vendor.card == null) return false;
```

Comment:
also you can check `if (vendor.available != true) return false;` after the available variable change mentioned above

Reply:
done